lib/puppet/provider/package/chocolatey.rb

```diff
@@ -52,6 +55,10 @@ def self.chocolatey_command
     chocopath
   end
 
+  def self.chocolatey_relocation_in_progress?
+     ENV['ChocolateyInstall'].downcase == 'c:\chocolatey' && file_exists?('C:\ProgramData\chocolatey\bin\choco.exe')
+  end
```

Comment:
We can just extract that logic to `puppet_x\chocolatey\chocolatey_install` and then use that in the fact and here.